### ISSUE TITLE
Create a composed Link component

### ISSUE THREAD

**Issue by samrichca** [icgc-argo/platform-ui#2402]
- Make a `<Link />` component that can handle internal and external links, and buttons that act like links.
- External links should use an HTML anchor tag wrapped around a uikit `Button` or `Link`. They should have the `target="_blank"` attribute.
- Internal links should use a `next/link` component wrapped around a uikit `Button` or `Link`. 
- Replace Link, NextLink, and HTML anchor tags throughout the repo with the new element.

Example usage:

```
<Link href={SOME_EXTERNAL_PATH>Click Me!</Link>
<Link href={urlJoin('/donor/', donorId}><Icon ... /></Link>
```

**Comment by caravinci** [comment]
While I support the button vs link abstraction, as it is a great idea, the `next/link` part of this may be difficult to implement. UiKit is not a `next.js`-specific app/plugin, so it doesn't necessarily have access to `next.js` components/api, and relies upon whatever app the integration is done.

**Comment by samrichca** [comment]
precisely, the next/link aspect would be handled in platform-UI not uikit

having a composed Link would save us from doing `<NextLink><Link>Link text</Link></NextLink>` all the time, is all

**Comment by caravinci** [comment]
Now I see what you mean. In that case, I may suggest changing the ticket to explain that more clearly (it reads to me as if `next/link` was intended as one of the parts handled by the composed Link. 👍

**Comment by samrichca** [comment]
next/link would be handled by the new Link, but only on the nextjs side of things
